testing/trino-testing/src/main/java/io/trino/testing/AbstractTestEngineOnlyQueries.java

```diff
@@ -393,7 +393,8 @@ public void testVarcharCharComparison()
                 "   CAST(' ' AS varchar(3)), " +
                 "   CAST('  ' AS varchar(3)), " +
                 "   CAST('   ' AS varchar(3))) t(x) " +
-                "WHERE x = CAST('  ' AS char(2))");
+                "WHERE x = CAST('  ' AS char(2))",
+                "VALUES (''), (' '), ('  '), ('   ')"); // h2 implicit coercion is not working correctly
```

Comment:
I did not understand this part.

Reply:
Got it.

Reply:
The h2 results changed.

@martint  take a look please